src/language-js/print/function.js

```diff
@@ -183,6 +192,53 @@ function printArrowFunctionExpression(path, options, print, args) {
 
   const body = path.call((bodyPath) => print(bodyPath, args), "body");
 
+  // For curried arrow functions like:
+  //  const foo =
+  //    (argument1) =>
+  //    (argument2) =>
+  //    (argument3) =>
+  //    (argument4) =>
+  //    (argument5) =>
+  //    (argument6) => ({
+  //      foo: bar,
+  //      bar: foo,
+  //    });
+  if (n.body.type === "ArrowFunctionExpression") {
+    const isCurriedChainingRoot = (path) =>
+      path.getParentNode().type !== "ArrowFunctionExpression";
+    const isCurriedChainingArrowFunction = (path) => {
+      if (isCurriedChainingRoot(path)) {
+        let arrowChainCount = 0;
+        let { body } = path.getNode();
+        while (body.type === "ArrowFunctionExpression") {
+          arrowChainCount++;
+          body = body.body;
+        }
+        return arrowChainCount > 3;
```

Comment:
Should return `ture` in loop, when count >=3, but do you think better use `2`?

Reply:
I think you misunderstood, I mean

```js
while () {
  if (arrowChainCount >= 2) return true
}
```

No need loop to the most inner one.

And I didn't mean `>= 3`.

Reply:
Ah I got it, thanks. I'll fix.
Btw which level do you think we should choose?

Reply:
I think `2`,

`const a = (b) => (c) => 1`, this is `2`, right ?

Reply:
Yes. I changed to choose level `2` at 731a12b. The commit updates some test cases, please check. I personally think we can approve those updates.